tests/admin_docs/tests.py

```diff
@@ -103,6 +116,31 @@ def test_view_detail_illegal_import(self):
         self.assertEqual(response.status_code, 404)
         self.assertNotIn("urlpatterns_reverse.nonimported_module", sys.modules)
 
+    @unittest.skipIf(six.PY2, "Python 2 doesn't support __qualname__.")
+    def test_view_detail_as_method(self):
+        """
+        #27018 - Ensure views that are methods can be displayed
+        """
+        url = reverse(
+            'django-admindocs-views-detail',
+            args=['django.contrib.admin.sites.AdminSite.index'],
+        )
+        response = self.client.get(url)
+        self.assertEqual(response.status_code, 200)
+
+    @unittest.skipUnless(six.PY2, "Python 2 doesn't support __qualname__.")
+    def test_view_detail_incorrect_method_path(self):
+        """
+        #27018 - Views that are methods can't be made to work on python 2
+        but should raise a 404
+        """
+        url = reverse(
+            'django-admindocs-views-detail',
+            args=['django.contrib.admin.sites.index'],
```

Comment:
Ah of course, it would work equally well merged into the previous test with your suggested logic. Thanks!